### PR TITLE
NPC agents in pilot's lounge

### LINKED ISSUE #34: Add pilot in lounge system
- Pilots, when not on a mission need to be chilling in the lounge
- When the mission starts, the pilot should disappear

### PR DESCRIPTION
**Pull request by maxwellpark**

### CODE REVIEW

**Comment by maxwellpark** [comment]
This is pretty experimental lol. No doubt better ways to do it, but I like being able to put this script on any plane object and it will automatically calculate the area they're restricted to.

Easier to see what's going on if you have gizmos turned on ingame.
The sphere gizmos show the boundary; the wire cube gizmos show the corresponding coloured npc's next destination. 

Upon reaching their destination, pilots wait for a period of time in some range. Then they move to their destination.
At the moment, their next destination is randomly generated.
But we could introduce an array of destination coordinates if we wanted them to go to specific places in the lounge, and then perhaps cue animations at those destinations (like eating/drinking).

Also, I haven't done any npc animation yet, they are always in down idle state.

File: Assets/Scripts/NPC/NPCAgent.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public void Wait()
+    {
+        isWaiting = true;
+    }
+
+    private void Update()
+    {
+        if (isWaiting)
+        {
+            timer--;
```
Comment (ambid17) [inline Assets/Scripts/NPC/NPCAgent.cs]: timer -= Time.DeltaTime

File: Assets/Scripts/NPC/NPCAgent.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public void Wait()
+    {
+        isWaiting = true;
+    }
+
+    private void Update()
+    {
+        if (isWaiting)
```
Comment (ambid17) [inline Assets/Scripts/NPC/NPCAgent.cs:40]: Doesn't stop when game is paused

File: Assets/Scripts/NPC/AIBoundingPlane.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    ///     Unity planes have an 11x11 grid of points.
+    ///     Index 0, 10, 110, and 120 are the four corners of the plane.
+    ///     The first two elements are the near corners, and the 
+    ///     last two are the far corners (depending on perspective).
+    ///     
+    ///     We need to get the world position via TransformPoint.
+    /// </summary>
+    /// <param name="vertices"> The raw vertices of the plane's mesh. </param>
+    /// <returns> The four corners of the bounding plane. </returns>
+    private List<Vector3> GetBoundaryVertices(Vector3[] vertices)
```
Comment (ambid17) [inline Assets/Scripts/NPC/AIBoundingPlane.cs:27]: this is NOT a long term solution

**Review by ambid17 — APPROVED** [review]

**Pull request merged**